integration_test/deployment-rds.sh

```diff
@@ -1,36 +1,56 @@
 #!/bin/bash
 set -ex
 
-export CUMULUS_AWS_ACCESS_KEY_ID=$bamboo_CUMULUS_AWS_ACCESS_KEY_ID
-export CUMULUS_AWS_SECRET_ACCESS_KEY=$bamboo_CUMULUS_AWS_SECRET_ACCESS_KEY
+export AWS_ACCESS_KEY_ID=$bamboo_CUMULUS_AWS_ACCESS_KEY_ID
+export AWS_SECRET_ACCESS_KEY=$bamboo_CUMULUS_AWS_SECRET_ACCESS_KEY
 export AWS_DEFAULT_REGION=$bamboo_CUMULUS_AWS_DEFAULT_REGION
 
 #remove old files from bamboo as they throw error
 rm *.tf
 #deploy the S3 buckets and dynamoDB table first
-git clone --branch develop --single-branch https://github.com/nasa/cumulus-orca.git && cd integration_test
+git clone --branch ${bamboo_BRANCH_NAME} --single-branch https://github.com/nasa/cumulus-orca.git && cd integration_test
+echo "Cloned Orca, branch ${bamboo_BRANCH_NAME}"
 #replace prefix with bamboo prefix variable
 sed -e 's/PREFIX/'"$bamboo_PREFIX"'/g' buckets.tf.template > buckets.tf
 
-if ! aws s3 cp s3://$bamboo_PREFIX-tf-state/buckets-tf/terraform.tfstate .;then
-    echo "terraform state file not present in S3. Creating the buckets and dynamoDB table..."
+if aws s3api head-bucket --bucket ${bamboo_PREFIX}-tf-state;then
+    echo "terraform state bucket already present. Using existing state file"
 else
-    echo "Terraform state file found in S3 bucket. Using S3 remote backend"
+    echo "Something went wrong when checking terraform state bucket. Creating ..."
+    aws s3api create-bucket --bucket ${bamboo_PREFIX}-tf-state  --region ${bamboo_AWS_DEFAULT_REGION} --create-bucket-configuration LocationConstraint=${bamboo_AWS_DEFAULT_REGION}
+    
+    aws s3api put-bucket-versioning \
+    --bucket ${bamboo_PREFIX}-tf-state \
+    --versioning-configuration Status=Enabled
+
+    aws dynamodb create-table \
+      --table-name ${bamboo_PREFIX}-tf-locks \
+      --attribute-definitions AttributeName=LockID,AttributeType=S \
+      --key-schema AttributeName=LockID,KeyType=HASH \
+      --billing-mode PAY_PER_REQUEST \
+      --region ${bamboo_AWS_DEFAULT_REGION}
 fi
```

Comment:
we can also create other buckets here instead of using another terraform file. In that case, we will not need buckets.tf.template. 
```
#create ORCA buckets
aws s3api create-bucket --bucket ${bamboo_PREFIX}-level0 --region ${bamboo_AWS_DEFAULT_REGION} --create-bucket-configuration LocationConstraint=${bamboo_AWS_DEFAULT_REGION}
aws s3api create-bucket --bucket ${bamboo_PREFIX}-internal --region ${bamboo_AWS_DEFAULT_REGION} --create-bucket-configuration LocationConstraint=${bamboo_AWS_DEFAULT_REGION}
aws s3api create-bucket --bucket ${bamboo_PREFIX}-private --region ${bamboo_AWS_DEFAULT_REGION} --create-bucket-configuration LocationConstraint=${bamboo_AWS_DEFAULT_REGION}
aws s3api create-bucket --bucket ${bamboo_PREFIX}-protected --region ${bamboo_AWS_DEFAULT_REGION} --create-bucket-configuration LocationConstraint=${bamboo_AWS_DEFAULT_REGION}
aws s3api create-bucket --bucket ${bamboo_PREFIX}-public --region ${bamboo_AWS_DEFAULT_REGION} --create-bucket-configuration LocationConstraint=${bamboo_AWS_DEFAULT_REGION}
``

Reply:
I vote we stick with centralizing in TF as much as possible, as that will make cleanup easier.

Reply:
fair enough.